x-pack/plugins/apm/public/components/app/Settings/AgentConfigurations/AgentConfigurationCreateEdit/SettingsPage/SettingsPage.tsx

```diff
@@ -37,7 +37,7 @@ import { useUiTracker } from '../../../../../../../../observability/public';
 import { SettingFormRow } from './SettingFormRow';
 import { getOptionLabel } from '../../../../../../../common/agent_configuration/all_option';
 
-function removeEmpty<T>(obj: T): T {
+function removeEmpty(obj: { [key: string]: any }) {
```

Comment:
You might be able to tighten this up like:
```suggestion
function removeEmpty(obj: Record<string, string>): Record<string, string> {
```

It requires that this is also updated (`setting_definitions/types.d.ts`):
```diff
-export type SettingValidation = t.Type<any, string, unknown>;
+export type SettingValidation = t.Type<string, string, unknown>;
```